### ISSUE TITLE
Large, large-v1, and large-v2 models always crash on Windows

### ISSUE THREAD

**Issue by DKWoods** [guillaumekln/faster-whisper#532]
This issue arises on Windows 11 using Python 3.11.6.  I do not see the same issue on macOS.

Whenever I use any of the 3 large models, the transcription fails.  I've tried multiple files in multiple formats, but the results are always the same.

`
    import logging, os
    import faster_whisper

    logging.basicConfig()
    logging.getLogger("faster_whisper").setLevel(logging.DEBUG)

    print('faster_whisper version:', faster_whisper.version.__version__)

    # CUSTOMIZE these two values!
    dataFilename = 'E:\\Videos\\Ellen Feiss.mp4'
    modelDirectory = 'D:\\(username)\\Documents\\faster_whisper_models'

    print('dataFilename "{0}" exists: {1}'.format(dataFilename, os.path.exists(dataFilename)))

    # models = faster_whisper.available_models()
    models = ('large-v1',)  # ('medium',)  ('large',)

    for modelSize in models:

        modelDir = os.path.join(modelDirectory, modelSize)

        print('Model: {0}  Dir exists: {1}'.format(modelSize, os.path.isdir(modelDir)))

        if not os.path.isdir(modelDir):

            print()
            print('Downloading model "{0}" to "{1}"'.format(modelSize, modelDir))
        
            faster_whisper.download_model(modelSize, cache_dir=modelDir)
        
            print('Download completed.')
          
        print()
        print('About to create Model.')

        model = faster_whisper.WhisperModel(modelSize, device="auto", download_root=modelDir)
            
        print('Model created.  About to transcribe.')
        print()

        compression_ratio = 2.4  # 2.4  20.0
        log_prob = -1  # -1  -300.0
            
        (segments, info) = model.transcribe(dataFilename, word_timestamps=True,
                                            compression_ratio_threshold=compression_ratio,
                                            log_prob_threshold=log_prob)

        print("Detected language '%s' with probability %f" % (info.language, info.language_probability))
        print('duration:', info.duration)
        print()
    
        print('Transcription process initiated.  Processing Results.')
        print()

        segmentCounter = 0
        wordCounter = 0
            
        for segment in segments:

            print('[%.2fs -> %.2fs] "%s"' % (segment.start, segment.end, segment.text))

            segmentCounter += 1

            for word in segment.words:
            
                print('[%.2fs -> %.2fs] "%s"  (%4.2f)' % (word.start, word.end, word.word, word.probability))

                wordCounter += 1

        print()
        print('------------------------------------------------------------------------------')
        print()
        print(modelSize, 'complete with {0} segments and {1} words.'.format(segmentCounter, wordCounter))
`
This works as I expect with the medium model and smaller. 

With any of the large models, the first sign of trouble I get is that info.language_probability returns "nan".

I get two variations of DEBUG messages:

`DEBUG:faster_whisper:Compression ratio threshold is not met with temperature 0.0 (18.666667 > 2.400000)
DEBUG:faster_whisper:Log probability threshold is not met with temperature 0.0 (-291.128876 < -1.000000)`

The first of those messages is usually repeated multiple times with different temperature values.  The "out of bounds" values stay the same (18.67 and -291.1) regardless of the file I am working with.  If I adjust the compression_ratio_threshold and log_prob_threshold values, I can make the DEBUG messages go away, but still get no segments when using the large models.  These values do not seem to affect the medium or smaller models.

I always get 0 segments returned with the large models, even when I adjust to avoid the DEBUG messages.

**Comment by lucasbr15** [comment]
I encountered a similar issue when using the v2 large model on Faster Whisper with Windows 11 and Python 3.9.9. Sometimes, the system crashes and restarts. However, this never happened on Windows 10.

**Comment by DKWoods** [comment]
I see that if I change the WhisperModel device from "auto" to "cpu", the large models suddenly work!

**Comment by DKWoods** [comment]
Sorry about that.  I was trying to put together the post and figure out its fuller dynamics at the same time.

The crash occurred on Windows when the WhisperModel device was "auto" or "cuda" (on my system with an NVidia card) when a non-0 temperature value was used when I tried to load the model for a second time.  You can see in the code above that I originally created a list of all available models and tried to iterate through them, but the program always crashed on the second iteration of the models loop.  I found this mentioned on another issue which suggested setting temperature to 0, which works.  Changing device to "cpu" also prevents the crash.

So if I set the temperature to 0, then the models fail rather than crash.  If I set the device to "cpu", I get segment data.  I should have adjusted the issue title.  Still, returning no segment data for files that other models properly transcribe is a significant problem.  I'm still trying to figure out how to maximize accuracy with reasonable speed with all these settings.

**Comment by Purfview** [comment]
Crash could be because "out of memory" issue, post VRAM size of your GPU.

**Comment by DKWoods** [comment]
VRAM is 5979 MB.  I don't get any DEBUG error messages.

Testing with 30 second audio files, but will need to be able handle files as long as 2 hours in production.

**Comment by DKWoods** [comment]
Thanks for your feedback and assistance.

Since I am writing an application for non-technical users on arbitrary hardware, I've concluded that processing using the GPU can be problematic.  I can get pretty reliable crashes when trying to specify a model for the second time when the GPU is involved.  (In particular, deleting the existing model crashes, as does trying to create a new model for a new transcription without deleting the previous model.)  I also get problematic results at times using the GPU that I don't really see using the CPU.  (eg. an audio file with 73 words reports 217 words with one word repeated many times.)  I had many more DEBUG messages about compression_ratio and log_probability_threshold values.  I did not have any luck improving things manipulating compute_type or temerature (as recommended in a different issue's discussion).

For my purposes, if I stick to device="cpu" and compute_type="float32", I get really good results and very reliable performance on both Windows 11 and macOS.  I might not get those results as fast as I might otherwise, but not having crashes and not having to repeat transcription calls because of problematic results is a good trade-off for my purposes.  

Thanks for sharing this great tool.
